ql/math/statistics/incrementalstatistics.hpp

```diff
@@ -149,25 +138,10 @@ namespace QuantLib {
         void reset();
         //@}
      private:
-       typedef boost::accumulators::accumulator_set<
-           Real,
-           boost::accumulators::stats<
-               boost::accumulators::tag::count, boost::accumulators::tag::min,
-               boost::accumulators::tag::max,
-               boost::accumulators::tag::weighted_mean,
-               boost::accumulators::tag::weighted_variance,
-               boost::accumulators::tag::weighted_skewness,
-               boost::accumulators::tag::weighted_kurtosis,
-               boost::accumulators::tag::sum_of_weights>,
-           Real> accumulator_set;
-        accumulator_set acc_;
-        typedef boost::accumulators::accumulator_set<
-            Real, boost::accumulators::stats<
-                      boost::accumulators::tag::count,
-                      boost::accumulators::tag::weighted_moment<2>,
-                      boost::accumulators::tag::sum_of_weights>,
-            Real> downside_accumulator_set;
-        downside_accumulator_set downsideAcc_;
+        class accumulator_set;
+        class downside_accumulator_set;
+        accumulator_set* acc_;
```

Comment:
I'd go for `unique_ptr` otherwise you need a copy constructor and operator for this to be safe.

Reply:
Agreed.